posthog/middleware.py

```diff
@@ -270,3 +276,80 @@ def __call__(self, request: HttpRequest):
                 reset_query_tags()
         response: HttpResponse = self.get_response(request)
         return response
+
+
+class CaptureMiddleware:
+    """
+    Middleware to serve up capture responses. We specifically want to avoid
+    doing any unnecessary work in these endpoints as they are hit very
+    frequently, and we want to provide the best availability possible, which
+    translates to keeping dependencies to a minimum.
+    """
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+        middlewares: List[Any] = []
+        # based on how we're using these middlewares, only middlewares that
+        # have a process_request and process_response attribute can be valid here.
+        # Or, middlewares that inherit from `middleware.util.deprecation.MiddlewareMixin` which
+        # reconciles the old style middleware with the new style middleware.
+        for middleware_class in (
+            PrometheusBeforeMiddleware,
+            SecurityMiddleware,
+            CorsMiddleware,
+            PrometheusAfterMiddleware,
+        ):
+            try:
+                # Some middlewares raise MiddlewareNotUsed if they are not
+                # needed. In this case we want to avoid the default middlewares
+                # being used.
+                middlewares.append(middleware_class(get_response=None))
```

Comment:
I could potentially simulate what django's get_response does so we can get rid of the old-style limitation 🤔 . But didn't seem necessary right now, and I'd like to avoid another rabbit hole :P